### PR TITLE
[SW-2559] Expose Scoring History and Variable Importances on H2OMOJOModel

### PR DESCRIPTION
**Pull request by mn-mikke**
Jira ticket: https://h2oai.atlassian.net/browse/SW-2559

### CODE REVIEW

File: py-scoring/src/ai/h2o/sparkling/ml/params/H2OMOJOModelParams.py [hunk 1]
```diff
@@ -46,6 +46,12 @@ def getTrainingParams(self):
     def getModelCategory(self):
         return self._java_obj.getModelCategory()
 
+    def getScoringHistory(self):
```
Comment (michalkurka) [inline py-scoring/src/ai/h2o/sparkling/ml/params/H2OMOJOModelParams.py:49]: is it documented anywhere what should users expect on the output?

Comment (mn-mikke) [reply py-scoring/src/ai/h2o/sparkling/ml/params/H2OMOJOModelParams.py:49]: Currently, It's not documented. I will improve that.

File: py-scoring/src/ai/h2o/sparkling/ml/params/H2OMOJOModelParams.py [hunk 2]
```diff
@@ -46,6 +46,12 @@ def getTrainingParams(self):
...
+        return H2OTypeConverters.scalaToPythonDataFrame(self._java_obj.getScoringHistory())
+
+    def getVariableImportances(self):
+        return H2OTypeConverters.scalaToPythonDataFrame(self._java_obj.getVariableImportances())
+
```
[the remark this answers is not in the document]
Comment (mn-mikke) [reply py-scoring/src/ai/h2o/sparkling/ml/params/H2OMOJOModelParams.py:54]: Great suggestion, regarding naming conventions. We try to follow Spark terminology if possible. variables -> features, target -> label. 

Regarding the type, I would keep using DataFrames. Spark algorithms get all features as one Vector columns and you can identify an individual feature only by its position. The same analogy is used for `featureImportances`

SW algorithms accept features in multiple columns and IMHO it's great benefit to associate features with their names. H2O-3 `variableImportances` also offer more values for a particular feature (`Relative Importance`, `Scaled Importance`, `Percentage`)

**Comment by mn-mikke** [comment]
> When people are training CV models, how do they have access to scoring history of the CV models?

As discussed, it's not currently possible

> Also how do they access CV metrics, training metrics and validation metrics? 
```
scala> model.getTrainingMetrics()
res2: Map[String,Double] = Map(Logloss -> 0.5407563235669484, Gini -> 0.5738086864779322, RMSE -> 0.42418099556613403, MSE -> 0.17992951699947662, R2 -> 0.24477535552017637, AUC -> 0.7869043432389661)

scala> model.getValidationMetrics()
res3: Map[String,Double] = Map()

scala> model.getCrossValidationMetrics()
res4: Map[String,Double] = Map()
```

**Review by michalkurka — APPROVED** [review]

**Pull request merged**